int32 x
int32 y
int8 wallLeft
int8 wallUp
int8 wallDown
int8 wallRight
